packages/snap/src/index.ts

```diff
@@ -64,8 +64,24 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ origin, request }) =>
   }
 }
 
+/**
+ * Gets the connected Provider.
+ * 
+ * @dev The ethereum object does not expose the RPC URL for non Ethereum networks,
+ * however we are able to get chainId, which we use to resolve
+ * provider URL.
+ * @returns 
+ */
+async function getConnectedProvider(): Promise<any> {
+  const provider = new ethers.providers.Web3Provider(ethereum as any)
```

Comment:
Casting the ethereum provider to any here seems a little strange. Have you tried the `eth_chainId` RPC method? https://docs.metamask.io/wallet/get-started/detect-network